### DISCUSSION TITLE
Ask your questions here :)

### DISCUSSION THREAD

**Discussion by sosna** [sdmx-twg/sdmx-rest#174] [Announcements]
Welcome to the `Discussions` area for the SDMX REST specification!

Feel free to ask any question about the SDMX API. The maintainers of the specification will do their best to answer your questions swiftly, but please bear in mind that most of us are not working full time on SDMX topics ;-).

**Comment by buhaiovos** [comment]
Hi :)
I have a question related to the structure retrieval API (https://github.com/sdmx-twg/sdmx-rest/blob/master/doc/structures.md), specifically the `descendants` value of query parameter  `references`. Let's say our target artefact is data structure definition. When client resolves its descendants, should they expect all codelists which are referenced from the concepts schemes corresponding to data structure definition, or only those codelists which are referenced from the concepts used in DSD components? It seems that the docs above suggest that all codelists should be resolved regardless of the DSD components (i.e. all references of references), but it would be great if you could confirm that assumption.
Many thanks!

**Comment by agent96** [comment]
Hi, descendants resolves down the tree fully - so a query for a DSD will will return all Concepts and Codelists, including Codelists that are referenced by the Concepts

**Comment by egreising** [comment]
But I understand that the question is if codelists from other concepts in the ConceptScheme but not referenced in the DSD, and the answer should be No. Only codelists of concepts referenced by the DSD. The DSD references Concepts, not ConceptSchemes...

**Reply by buhaiovos** [reply]
Indeed, that is what I meant. So, in this situation distinct concepts should be taken into consideration and not the whole concept scheme which holds these concepts? Then, resolving references of those **concepts** will bring corresponding codelists.

**Reply by agent96** [reply]
Only if you include the parameter references=referencepartial - you can read the behaviour under the 'detail' parameter here:
https://github.com/sdmx-twg/sdmx-rest/blob/master/doc/structures.md

If you do not include this parameter, then you will get all Concepts back in the Concept Scheme and all Codelists back that are referenced by the Concepts regardless of what the DSD uses

**Reply by buhaiovos** [reply]
I see, now it makes sense. When `details`, say, `full` I will get all concepts from the corresponding concept scheme and hence all related codelists. But if it's `referencepartial` - stripped out "version" of the concept scheme is returned then (with `isPartial=true`), and as the result the number of referenced codelists also decreases.

**Reply by buhaiovos** [reply]
@agent96, @egreising, Thanks for your answers!

**Comment by KARINA115116** [comment]
Hi All,
Please help to understand what is wrong in the address of ECB exchange rates API?

**=IMPORTHTML("https://sdw.ecb.europa.eu/browseTable.do?org.apache.struts.taglib.html.TOKEN=5b56c9bf58355dbe1a2972c1cf2336be&df=true&MAX_DOWNLOAD_SERIES=500&DATASET=0&org.apache.struts.taglib.html.TOKEN=af54c2f2da88ee8e34ff55b4d3b5fa9f&org.apache.struts.taglib.html.TOKEN=6fff54088d5ec13458395da4bdad561b&node=qview&SERIES_MAX_NUM=50&SERIES_KEY=120.EXR.D.USD.EUR.SP00.A&activeTab=EXR&start=31-12-2018&end=&submitOptions.x=0&submitOptions.y=0&trans=N&q=&type=","table",7)**

The error is **Imported content does not have query with given index.**

Thanks

**Reply by dosse** [reply]
Hello @KARINA115116, unfortunately, the SDMX-TWG cannot answer you on this question, because it is about a specific implementation rather than the SDMX technical standard itself. Please contact the ECB directly on its own web site. Thank you for your comprehension.

**Comment by LuchiLucs** [comment]
I wish to educate myself how can I query data, but I am confused by the many required and optional parameters of the exposed APIs as well as what are the latest supported version/standard and what is going to be dismessed.

For instance, for the sake of a simply example, being italian I wish to retrieve the monthly index of inflation from the Istat site which from a UI interface let me compute automatically the query:
https://esploradati.istat.it/SDMXWS/rest/data/IT1,169_745_DF_DCSP_FOI1B2015_1,1.0/M..55../ALL/?detail=full&startPeriod=2016-01-01&endPeriod=2025-01-31&dimensionAtObservation=TIME_PERIOD

How can I traslate this to start playing with the swagger UI?
Thanks
